tests/integration/steps/integration_steps.go

```diff
@@ -0,0 +1,622 @@
+	// Check that we have the right number of rows
+	g(t).Expect(output).Should(gomega.ContainSubstring(fmt.Sprintf("(%d rows)", nbRows)))
+}
+
+func CreateCassandraTable(t *testing.T, tableName, keyspaceName string) {
+	runCassandraQueryAndGetOutput(t, fmt.Sprintf("CREATE KEYSPACE IF NOT EXISTS %s with replication = {'class':'SimpleStrategy', 'replication_factor':1};", keyspaceName))
+	runCassandraQueryAndGetOutput(t, fmt.Sprintf("CREATE TABLE IF NOT EXISTS %s.%s(id timeuuid PRIMARY KEY, val text);", keyspaceName, tableName))
+}
+
+func LoadRowsInTable(t *testing.T, nbRows int, tableName, keyspaceName string) {
+	for i := 0; i < nbRows; i++ {
+		runCassandraQueryAndGetOutput(t, fmt.Sprintf("INSERT INTO %s.%s(id,val) values(now(), '%d');", keyspaceName, tableName, i))
+	}
+}
+
+// Medusa related functions
+func CreateMedusaSecretWithFile(t *testing.T, secretFile string) {
+	home, _ := os.UserHomeDir()
+	medusaSecretPath, _ := filepath.Abs(strings.Replace(secretFile, "~", home, 1))
+	kubectlOptions := k8s.NewKubectlOptions("", "", namespace)
+	k8s.KubectlApply(t, kubectlOptions, medusaSecretPath)
+}
+
+func PerformBackup(t *testing.T, backupName string) {
+	backupChartPath, err := filepath.Abs("../../charts/backup")
+	g(t).Expect(err).To(gomega.BeNil())
+
+	helmOptions := &helm.Options{
+		SetValues: map[string]string{
+			"name":                     backupName,
+			"cassandraDatacenter.name": "dc1",
+		},
+		KubectlOptions: k8s.NewKubectlOptions("", "", namespace),
+	}
+	// Start backup
+	helm.Install(t, helmOptions, backupChartPath, backupName)
+
+	// Wait for the backup to be finished
+	g(t).Eventually(func() int {
+		return backupIsFinished(t, backupName)
+	}, 2*time.Minute, 5*time.Second).Should(gomega.Equal(1))
+}
+
+func backupIsFinished(t *testing.T, backupName string) int {
+	var nodes []string
+	log.Printf("Checking if backup %s is finished...", backupName)
+	kubectlOptions := k8s.NewKubectlOptions("", "", namespace)
+	output, err := k8s.RunKubectlAndGetOutputE(t, kubectlOptions, "get", "cassandrabackup", backupName, "-o", "jsonpath='{.status.finished}'")
+	if err == nil && len(output) > 0 {
+		json.Unmarshal([]byte(strings.ReplaceAll(output, "'", "")), &nodes)
+	}
+	return len(nodes)
+}
+
+func RestoreBackup(t *testing.T, backupName string) {
+	restoreChartPath, err := filepath.Abs("../../charts/restore")
+	g(t).Expect(err).To(gomega.BeNil(), "Couldn't find the absolute path for restore charts")
+
+	helmOptions := &helm.Options{
+		SetValues: map[string]string{
+			"backup.name":              backupName,
+			"cassandraDatacenter.name": "dc1",
+			"name":                     "restore-test",
+		},
+		KubectlOptions: k8s.NewKubectlOptions("", "", namespace),
+	}
+	helm.Install(t, helmOptions, restoreChartPath, "restore-test")
+
+	// Restore should trigger an update of the pod through cass-operator
+	g(t).Eventually(func() string {
+		return readCassandraOperatorProgress(t)
+	}, RETRY_TIMEOUT, 5*time.Second).Should(gomega.Equal("Updating"))
+
```

Comment:
Similar with backups, the CassandraRestore object has a `.status.finishTime` property that is set after the restore is complete. This happens after the CassandraDatacenter comes back to the ready state.

Reply:
Gotcha, will change the code accordingly.

Reply:
Done 👍